crates/worker/src/partition_processor_manager/mod.rs

```diff
@@ -702,6 +779,55 @@ impl PartitionProcessorManager {
         }
     }
 
+    fn spawn_create_snapshot_task(
+        &mut self,
+        partition_id: PartitionId,
+        result_sender: oneshot::Sender<SnapshotResult>,
+        archived_lsn_sender: watch::Sender<Option<Lsn>>,
+    ) {
+        if let Entry::Vacant(entry) = self.pending_snapshots.entry(partition_id) {
+            let config = self.updateable_config.live_load();
+
+            let snapshot_base_path = config.worker.snapshots.snapshots_dir(partition_id);
+
+            let create_snapshot_task = SnapshotPartitionTask {
+                cluster_name: config.common.cluster_name().into(),
+                node_name: config.common.node_name().into(),
+                partition_id,
+                partition_store_manager: self.partition_store_manager.clone(),
+                snapshot_base_path,
+                archived_lsn_sender,
+            };
+
+            let snapshot_span = tracing::info_span!("create-snapshot");
+            let spawn_task_result = restate_core::task_center().spawn_unmanaged(
+                TaskKind::PartitionSnapshotProducer,
+                "create-snapshot",
+                Some(partition_id),
+                async move { create_snapshot_task.create_snapshot().await }
+                    .instrument(snapshot_span),
```

Comment:
If you want the instrumentation closer to the code, then you could add `#[instrument(...)]` to the `create_snapshot` method instead of assigning the span manually here.